src/main/java/com/ruralsourcing/WeatherBot/service/Weatherbot.java

```diff
@@ -16,22 +17,22 @@
      *
      * @param args ignored
      */
-    public static void main(String[] args){
-
-        Gson gson = new Gson();
+    public static void main(String[] args) throws IOException {
 
-        String httpResponse ="";
+        //new object to hold api tokens from the properties file
+        //default constructor reads API tokens from config.properties
+        PropertyIO properties = new PropertyIO();
 
-        try {
-            httpResponse = HttpCalls.request();
-        } catch(IOException e){
-            System.out.println(e);
-        }
+        //initialize the atomic reference holding the data
+        final AtomicReference<ResponseModel> ref = new AtomicReference<>();
 
-        //currentData holds the entire parsed JSON object
-        //It holds all current information as an array of Stations and their data
-        ResponseModel currentData = gson.fromJson(httpResponse, ResponseModel.class);
+        // Initialize timer thread that will update the data using API calls one time per hour
+        final Timer timer = new Timer("Timer", true);
+        timer.scheduleAtFixedRate(new ApiTimerTask(ref, properties.getMesonetApiToken()), 0, TimeUnit.HOURS.toMillis(1));
```

Comment:
To prevent the issue you mentioned earlier about timing you could pull the `new ApiTimerTask(ref, properties.getMesonetApiToken())` out into a variable, call `run()` to initialize it once, then put it into the timer task:

```java
final ApiTimerTask task = new ApiTimerTask(ref, properties.getMesonetApiToken());
// run once to ensure the reference is initialized
task.run();
final Timer timer = new Timer("Timer", true);
timer.scheduleAtFixedRate(task, 0, TimeUnit.HOURS.toMillis(1));
```

Alternatively, you could just do your first `run()` call in the constructor, which would also be fine and probably cleaner.